src/test/java/subway/LineTest.java

```diff
@@ -0,0 +1,227 @@
+package subway;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.restassured.RestAssured;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.Sql.ExecutionPhase;
+import subway.fixture.LineFixture;
+import subway.fixture.StationFixture;
+import subway.line.LineResponse;
+import subway.line.LineUpdateRequest;
+import subway.station.StationResponse;
+
+@Sql(value = "/truncate.sql", executionPhase = ExecutionPhase.BEFORE_TEST_METHOD)
+@DisplayName("지하철노선 관련 기능")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class LineTest {
+
+  @LocalServerPort
+  private int port;
+
+  @BeforeEach
+  void setUp() {
+    RestAssured.port = port;
+  }
+
+  /**
+   * When 노선을 생성하면
+   * Then 지하철 노선 목록 조회 시 생성한 노선을 찾을 수 있다
+   */
+  @DisplayName("노선을 생성한다.")
+  @Test
+  void createLineSuccess() {
+    // given
+    final var LINE_NAME = "신분당선";
+    final var upStation = StationFixture.createStation("강남역");
+    final var downStation = StationFixture.createStation("청계산입구역");
+
+    // when
+    Map<String, Object> params = new HashMap<>();
+    params.put("name", LINE_NAME);
+    params.put("color", "bg-red-600");
+    params.put("upStationId", upStation.getId());
+    params.put("downStationId", downStation.getId());
+    params.put("distance", 10);
+
+    final var response = RestAssured
+        .given().log().all()
+        .body(params)
+        .contentType(MediaType.APPLICATION_JSON_VALUE)
+        .when().post("/lines")
+        .then().log().all()
+        .extract();
+
+    final var line = LineFixture.getLines().stream()
+        .filter(it -> LINE_NAME.equals(it.getName()))
+        .findFirst();
+    final var stationIds = line.map(LineResponse::getStations)
+        .orElseGet(Collections::emptyList).stream()
+        .map(StationResponse::getId)
+            .collect(Collectors.toList());
+
+    // then
+    assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value());
+
+    // then
+    assertThat(line).isNotEmpty();
+
+    // then
+    assertThat(upStation.getId()).isIn(stationIds);
```

Comment:
```suggestion
    assertThat(LineFixture.getLines()).anyMatch(lineResponse -> lineResponse.getName().equals(LINE_NAME));

```

요런 방식도 사용가능하니 여러가지 assertion을 사용해보시면 좋을듯합니다!

Reply:
오 이거 직관적이네요! 👍